StandardLibraryEssentials/StandardTemplateLibrary/STLAlgorithms/task.md

```diff
@@ -0,0 +1,179 @@
+In addition to all that we have learned so far, the STL also provides a number of algorithms that can be used on a range of elements. These algorithms are implemented as functions that usually utilize iterators. In this task, we will look at some of the most commonly used algorithms. There are many more algorithms available in the STL, and you can find a complete list [here](https://en.cppreference.com/w/cpp/algorithm).
+We highly encourage you to explore the STL algorithms by yourself and try to use them in your code! They are useful and can save you a lot of time and effort.
+
+#### 1. `std::sort()`
+The [`std::sort()`](https://en.cppreference.com/w/cpp/algorithm/sort) algorithm is used to sort a range of elements. It is implemented as a function template of a `O(n log n)` sorting algorithm (most likely, [introsort](https://en.wikipedia.org/wiki/Introsort)), and the syntax is as follows:
+```cpp
+std::sort(start_iterator, end_iterator, comparator);
+```
+The first two arguments are iterators that define the range of elements to be sorted. The third argument is an optional comparator used to compare two elements. If no comparator is provided, the elements are sorted in ascending order by default. The comparator can be a function pointer, a lambda expression (small in-place function), or a functor.
+```cpp
+#include <iostream>
+#include <vector>
+#include <algorithm>
+#include <string>
+
+int main() {
+    // Sort a vector of integers in ascending order
+    std::vector<int> v1 = {5, 2, 4, 3, 1};
+    std::sort(v1.begin(), v1.end());
+    // v = {1, 2, 3, 4, 5}
+
+    // Sort a vector of integers in descending order
+    std::vector<int> v2 = {5, 2, 4, 3, 1};
+    std::sort(v2.begin(), v2.end(), std::greater<>());
+    // v = {5, 4, 3, 2, 1}
+
+    // Sort a vector of strings in ascending order by length
+    std::vector<std::string> v3 = {"abc", "a", "ab"};
+    std::sort(v3.begin(), v3.end(), [](const std::string &a, const std::string &b) {
+        return a.size() < b.size();
+    });
+    // v = {"a", "ab", "abc"}
+}
```

Comment:
One general notice for all code snippets like this used in this module. 
Please consider shortening them, and do not give the full code with all the includes, `main` declaration, etc.